Code/Legacy/CrySystem/SystemInit.cpp

```diff
@@ -896,16 +896,14 @@ void CSystem::InitLocalization()
     if (auto console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
     {
         AZ::CVarFixedString languageAudio;
-        if (auto result = console->GetCvarValue("g_languageAudio", languageAudio); result == AZ::GetValueResult::Success)
+        console->GetCvarValue("g_languageAudio", languageAudio);
+        if (languageAudio.size() == 0)
         {
-            if (languageAudio.size() == 0)
-            {
-                console->PerformCommand(AZStd::string::format("g_languageAudio %s", language.c_str()).c_str());
-            }
-            else
-            {
-                language.assign(languageAudio.data(), languageAudio.size());
-            }
+            console->PerformCommand(AZStd::string::format("g_languageAudio %s", language.c_str()).c_str());
+        }
```

Comment:
I take it the assumption here is that it will cause a trigger to occur that will change the language audio, so it doesn't have to also call  language.assign?

Reply:
This `PerformCommand` just sets the audio language cvar to whatever the general language is.  This is occurring before audio gems start getting initialized so later there's an explicit call to update the audio language.

The `language` here is just a local string that was set to the value of `g_language` cvar - the general language setting.  If it sees a value in `g_languageAudio` it writes that value to the `language` variable, then passes it to `OpenLanguageAudioPak`.  So either you're passing the value of `g_language` or `g_languageAudio` to `OpenLanguageAudioPak`.

Regardless, this is legacy Cry localization handling, with it's own assumptions about pak file names and locations.  It all needs to get scrapped anyways.  This update was just meant to fix it enough to propagate the setting to Wwise properly.